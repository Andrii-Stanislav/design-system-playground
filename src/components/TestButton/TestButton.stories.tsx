import type { Meta, StoryObj } from '@storybook/react';
import { TestButton } from './TestButton';

const meta: Meta<typeof TestButton> = {
  title: 'TestButton',
  component: TestButton,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof TestButton>;

export const Primary: Story = {
  args: {
    label: 'Button',
  },
};

export const Secondary: Story = {
  args: {
    label: 'Secondary Button',
  },
};
