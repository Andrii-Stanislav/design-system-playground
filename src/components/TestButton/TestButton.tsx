import React from 'react';

interface ButtonProps {
  label: string;
  onClick?: () => void;
}

export const TestButton: React.FC<ButtonProps> = ({ label, onClick }) => (
  <button onClick={onClick}>{label}</button>
);
