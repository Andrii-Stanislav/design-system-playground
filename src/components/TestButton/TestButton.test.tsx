import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TestButton } from './TestButton';
import { jest } from '@jest/globals';

describe('TestButton', () => {
  it('renders with the correct label', () => {
    render(<TestButton label="Click me" />);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });

  it('calls onClick handler when clicked', () => {
    const handleClick = jest.fn();
    render(<TestButton label="Click me" onClick={handleClick} />);

    fireEvent.click(screen.getByText('Click me'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('renders without onClick handler', () => {
    render(<TestButton label="Click me" />);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });
});
