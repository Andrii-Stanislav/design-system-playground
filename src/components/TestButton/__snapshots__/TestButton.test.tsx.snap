// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestButton renders correctly 1`] = `
<button>
  Click me
</button>
`;
